processor/internal/transformer/destination_transformer/embedded/warehouse/datatype.go

```diff
@@ -142,3 +148,58 @@ func convertToSliceIfViolationErrors(val any) any {
 	}
 	return val
 }
+
+func addDataAndMetadataForContextGeoEnrichment(tec *transformEventContext, data map[string]any, metadata map[string]string, key string, val any) error {
+	if geoLocation, ok := val.(enricher.Geolocation); ok {
+		if len(geoLocation.IP) > 0 {
```

Comment:
Instead of having an if block for each field, we can have a for loop

Reply:
This would require creating an array for key (`_ip`) and data (`geoLocation.IP`) as these are struct fields which is probably not required.